client/me/account/main.jsx

```diff
@@ -138,15 +145,29 @@ const Account = createReactClass( {
 	},
 
 	getUserSetting( settingName ) {
-		return this.props.userSettings.getSetting( settingName );
+		return (
+			get( this.props.unsavedUserSettings, settingName ) ??
+			this.getUserOriginalSetting( settingName )
+		);
 	},
 
 	getUserOriginalSetting( settingName ) {
-		return this.props.userSettings.getOriginalSetting( settingName );
+		return get( this.props.userSettings, settingName );
```

Comment:
What's the benefit of using `get()` here instead of just going with optional chaining?

FWIW we also have other instances of `get()` here, consider this question applicable to them, too.

Reply:
See my comment here: https://github.com/Automattic/wp-calypso/pull/49760#discussion_r571189849